plover/oslayer/osxkeyboardcontrol.py

```diff
@@ -285,18 +286,15 @@ def suppress_keyboard(self, suppressed_keys=()):
 # let's us iterate over full characters in the string.
 def characters(s):
     encoded = s.encode('utf-32-be')
-    characters = []
-    for i in xrange(len(encoded)/4):
+    for i in xrange(len(encoded) / 4):
         start = i * 4
         end = start + 4
         character = encoded[start:end].decode('utf-32-be')
         yield character
 
-native_utf16 = 'utf-16-le' if sys.byteorder == 'little' else 'utf-16-be'
-
 def set_string(event, s):
-    buf = s.encode(native_utf16)
-    CGEventKeyboardSetUnicodeString(event, len(buf) / 2, buf)
+    buf = Foundation.NSString.stringWithString_(s)
+    CGEventKeyboardSetUnicodeString(event, len(buf), buf)
```

Comment:
Actually, when I made an ord array, PyObjc yelled at me, telling me it needed a string as argument 3, so honestly this stuff feels a little bit like voodoo. I always sacrifice a lamb before touching `osxkeyboardcontrol.py`